File: online_judges/uva/p10474.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    //freopen("i","r",stdin);
    int n,Q,cnt=0;
    while(scanf("%d%d",&n,&Q)==2){
        if(n==0 && Q==0) break;
        cnt++;
        vector<int> mar(n);
        for(int i=0;i<n;i++) scanf("%d",&mar[i]);
        sort(mar.begin(),mar.end());
        printf("CASE# %d:\n",cnt);
        while(Q--){
            int x;
            scanf("%d",&x);
            int l = lower_bound(mar.begin(),mar.end(),x) - mar.begin();
            if(mar[l]==x) printf("%d found at %d",x,l+1);
            else printf("%d not found",x);
            printf("\n");
        }

    }
    return 0;
}

File: online_judges/uva/p10474.py
from bisect import *

cnt=0
while True:
    n,Q = input().split()
    n = int(n)
    Q = int(Q)
    if n==0 and Q==0:
        break
    cnt+=1
    a = input().split()
    A = [int(a[i]) for i in range(n)]
    q = [int(a[i]) for i in range(n,n+Q,1)]
    A.sort()
    print("CASE# ",cnt,":",sep='')
    for k in range(Q):
        i = bisect_left(A,q[k])
        if i==n or A[i]!=q[k]:
            print(q[k]," not found",sep='')
        else :
            print(q[k]," found at ",i+1,sep='')
